aspnetcore/fundamentals/routing.md

```diff
@@ -1,77 +1,326 @@
+
+The link generator can link to MVC/Razor Pages actions and pages via the following extension methods:
+
+* `GetPathByAction`
+* `GetUriByAction`
+* `GetPathByPage`
+* `GetUriByPage`
+
+An overload of these methods accept arguments that include the `HttpContext`. These methods are functionally equivalent to `Url.Action` and `Url.Page` but offer additional flexibility and options.
+
+The `GetPath*` methods are most similar to `Url.Action` and `Url.Page` in that they generate a URI containing an absolute path. The `GetUri*` methods always generate an absolute URI containing a scheme and host. The methods that accept an `HttpContext` generate a URI in the context of the executing request. The ambient route values, URL base path, scheme, and host from the executing request are used unless overridden.
+
+`LinkGenerator` is called with an address. Generating a URI occurs in two steps:
+
+1. An address is bound to a list of endpoints that match the address.
+1. Each endpoint's `RoutePattern` is evaluated until a route pattern that matches the supplied values is found. The resulting output is combined with the other URI parts supplied to the link generator and returned.
+
+The methods provided by `LinkGenerator` support standard link generation capabilities for any type of address. The most convenient way to use the link generator is through extension methods that perform operations for a specific address type.
+
+| Extension Method   | Description                                                         |
+| ------------------ | ------------------------------------------------------------------- |
+| `GetPathByAddress` | Generates a URI with an absolute path based on the provided values. |
+| `GetUriByAddress`  | Generates an absolute URI based on the provided values.             |
+
+> [!WARNING]
+> Pay attention to the following security implications of calling `LinkGenerator` methods:
+>
+> Use `GetUri*` extension methods with caution in an app configuration that doesn't validate the `Host` header of incoming requests. If the `Host` header of incoming requests isn't validated, untrusted request input can be sent back to the client in URIs in a view/page. We recommend that all production apps configure their server to validate the `Host` header against known valid values.
+>
+> Use `LinkGenerator` with caution in middleware in combination with `Map` or `MapWhen`. `Map*` changes the base path of the executing request, which affects the output of link generation. All of the `LinkGenerator` APIs allow specifying a base path. Always specify an empty base path to undo `Map*`'s affect on link generation.
+
+A few differences exist between endpoint routing in ASP.NET Core 2.2 or later and earlier versions of routing in ASP.NET Core:
```

Comment:
Should this be in a separate section that describes differences between the two frameworks? I wouldn't expect most users to need to know this.